src/main/java/htsjdk/tribble/index/IndexFactory.java

```diff
@@ -438,13 +438,24 @@ public FeatureIterator(final File inputFile, final FeatureCodec<FEATURE_TYPE, SO
                 throw new IllegalArgumentException("FeatureIterator input file cannot be null");
             }
             this.codec = codec;
-            this.inputFile = inputFile;
+
+            // We _MUST_ call getPathToDataFile here to preserve behavior of codecs that use parallel files.
+            // That is, codecs that rely on config files to point to other files to be indexed.
+            // This is CRITICALLY IMPORTANT to downstream tools.
+            final String file_path = codec.getPathToDataFile(inputFile.getAbsolutePath());
```

Comment:
code conventions for local variable names is intercaps
```suggestion
            final String filePath = codec.getPathToDataFile(inputFile.getAbsolutePath());
```